docs/metasploit-framework.wiki/dev/Setting-Up-a-Metasploit-Development-Environment.md

```diff
@@ -129,27 +127,83 @@ Congratulations! You have now set up a development environment and the latest ve
 
 ## Optional: Set up the REST API and PostgreSQL database
 
-The following optional section describes how to manually install PostgreSQL and set up the Metasploit database.  Alternatively, use our Omnibus installer which handles this more reliably.
+Installing the REST API and PostgreSQL is optional, and can be done in two ways.
+Recommended is to use the Docker approach, and fairly simple to do once you have docker installed on your
+system, [Docker Desktop][docker-desktop] is recommended, but not mandatory.
+On Linux systems, simply having docker-cli is sufficient.
+
+### Docker Installation
+
+**Make sure, you have docker available on your system: [Docker Installation Guide][docker-installation]**
 
-1. Confirm that the PostgreSQL server and client are installed:
+**Note**: Depending on your environment, these commands might require `sudo`
+
+* Pull the Docker container: `docker pull postgres:14`
+* Start the container:
+
+```bash
+docker run --rm -p 127.0.0.1:5432:5432 -e POSTGRESS_PASSWORD="mysecretpassword" postgres:14
+```
+
+Wait till the container is fully running.
+We need to make some changes in the container to be able to connect to it remotely using the `psql` client.
+
+* Open a shell to the running container: `docker exec -it "postgres" bash`
+* Install an editor: `sudo apt update && sudo apt install -y nano`
+* Edit `/var/lib/postgresql/data/postgresql.conf` and uncomment: `# password_encryption = scram-sha-256	# scram-sha-256 or md5`
+* Exit the shell on the container
+* Ensure the postgres client is installed on your host system
+* connect to the running container (the password is `mysecretpassword`): `psql -h 127.0.0.1 -U postgres`
+
+After this you should be able to connect with the `postgres` user using the password from the commands.
+We highly recommend to use the same credentials and use the `database.yml` below as example.
+With this approach you do not require the `./msfdb init` command. The standard Rails commands for creating
+the database are sufficient: `bundle exec rails db:create db:migrate db:seed`
+
+### database.yml
+
+```yaml
+development: &defaults
+  adapter: postgresql
+  database: metasploit_pro_development
+  username: postgres
+  password: mysecretpassword
+  host: 127.0.0.1
+  port: 5432
+  pool: 200
+  timeout: 5
+
+production:
+  <<: *defaults
+  database: metasploit_pro_production
+
+test:
+  <<: *defaults
+  database: metasploit_pro_test
+```
```

Comment:
This still looks very complicated to my eyes; I think this can just be closer to the manual postgres setup below:

````suggestion
* Start the postgres container:

```bash
docker run --rm -it -p 127.0.0.1:5433:5432 -e POSTGRES_PASSWORD="mysecretpassword" postgres:14
```

Wait till the postgres container is fully running.

* Configure the Metasploit database:

```
cd ~/git/metasploit-framework
./msfdb init --connection-string="postgres://postgres:mysecretpassword@127.0.0.1:5433/postgres"
```

* If the `msfdb init` command succeeds, then confirm that the database is accessible to Metasploit:

```bash
$ ./msfconsole -qx "db_status; exit"
```
````

Note that I also changed the bind port, added flags for `-it`, and fixed the `POSTGRESS_PASSWORD` typo in the docker command